server/src/test/java/org/jetbrains/bsp/bazel/server/diagnostics/DiagnosticsServiceTest.kt

```diff
@@ -500,6 +502,52 @@ class DiagnosticsServiceTest {
         diagnostics shouldContainExactlyInAnyOrder expected
     }
 
+    @Test
+    fun `should clear former diagnostics`() {
```

Comment:
I tried to add an unit test for that scenario in https://github.com/JetBrains/bazel-bsp/pull/381/commits/848ab4559a46413cf48ca2e077f3774b55d66a58
but it seems like we may want to write an e2e test, because the behavior involves upstream layers like `BazelBspCompilationManager` and `BepServer` 🤔

Reply:
it'd be nice to have e2e test for it (and we will have them soon or later) but for now we dont have 'framework' to check notifications, so it'd be very difficult to test it

anyway in this case the only missing thing is to check the state - is it empty or not after partial clean

Reply:
Added unit test for verifying the state after partial clean https://github.com/JetBrains/bazel-bsp/pull/381/commits/2122da700592ae94ccbc352fd4f7c111d08b9b86 :+1: